src/Analyzer/Passes/QueryAnalysisPass.cpp

```diff
@@ -2465,23 +2465,19 @@ QueryTreeNodePtr QueryAnalyzer::tryResolveIdentifierFromTableExpression(const Id
                 result_column = it->second;
         }
 
-        QueryTreeNodePtr result_expression;
+        QueryTreeNodePtr result_expression = result_column;
         bool clone_is_needed = true;
 
         String table_expression_source = table_expression_data.table_expression_description;
         if (!table_expression_data.table_expression_name.empty())
             table_expression_source += " with name " + table_expression_data.table_expression_name;
 
-        if (!match_full_identifier && compound_identifier)
+        if (result_column && !match_full_identifier && compound_identifier)
```

Comment:
Maybe `!result_column`? Why would we change the resolve result if we've already found a column?

Reply:
We match only first identifier part, if identifier is compound and we does not match full identifier, we must resolve identifier from compound expression.